knox/views.py

```diff
@@ -38,20 +41,41 @@ def post(self, request, format=None):
                     {"error": "Maximum amount of tokens allowed per user exceeded."},
                     status=status.HTTP_403_FORBIDDEN
                 )
+        show_token_ttl = self.get_ttl_visibility()
         token_ttl = self.get_token_ttl()
         token = AuthToken.objects.create(request.user, token_ttl)
         user_logged_in.send(sender=request.user.__class__,
                             request=request, user=request.user)
         UserSerializer = self.get_user_serializer_class()
-        if UserSerializer is None:
-            return Response({
-                'token': token
-            })
+
+        if UserSerializer is None and show_token_ttl is False:
+            return Response(
+                {
+                    'token': token
+                }
+            )
+        elif UserSerializer is None and show_token_ttl is True:
+            return Response(
+                {
+                    'token': token,
+                    'token_expiry': token_ttl
+                }
+            )
         context = self.get_context()
-        return Response({
-            'user': UserSerializer(request.user, context=context).data,
-            'token': token,
-        })
+        if show_token_ttl is True:
+            return Response(
+                {
+                    'user': UserSerializer(request.user, context=context).data,
+                    'token': token,
+                    'token_expiry': token_ttl,
+                }
+            )
+        return Response(
+            {
+                'user': UserSerializer(request.user, context=context).data,
+                'token': token
+            }
+        )
```

Comment:
- There is no need to check specifically for `True` and `False` any value that can be casted to boolean should be fine.

- The logic could be simplified and made more readable by doing:
```
if UserSerializer is None:
    resp = {...}
else:
    resp = {...}

if show_token_ttl:
    resp["token_expiry"] = token_ttl

return resp

Reply:
All good points and yes, it did feel a tad ugly to do those if-statements @johnraz, I will fix it asap :)

Reply:
@johnraz its done :) but hey, my naming tho, `data`? haha

Reply:
`data` is great 👍